compiler/natives/src/strconv/atoi.go

```diff
@@ -0,0 +1,31 @@
+// +build js
+
+package strconv
+
+import (
+	"github.com/gopherjs/gopherjs/js"
+)
+
+const maxInt32 float64 = 1<<31 - 1
+const minInt32 float64 = -1 << 31
+
+// Atoi returns the result of ParseInt(s, 10, 0) converted to type int.
+func Atoi(s string) (int, error) {
+	const fnAtoi = "Atoi"
+	if len(s) == 0 {
+		return 0, syntaxError(fnAtoi, s)
+	}
+	jsValue := js.Global.Call("Number", s, 10)
```

Comment:
I believe JS's `Number` constructor is far more forgiving than Go's `Atoi` implementation. Is it therefore reasonable to validate that `s` contains only Go-acceptable numbers?

Some examples of values that `Number()` accepts but `Atoi()` should not:

- "0x10"
- "10.2"
- "0b11"
- "0o10"

Reply:
Validate doesn't seem to have hurt perf at all -- pushing up the changes.

Reply:
Looks like a good improvement. And glad to hear it doesn't have a noticeable impact on performance!